springboot-jersey-app/delivery/src/main/java/com/wfsample/delivery/DeliveryController.java

```diff
@@ -0,0 +1,41 @@
+package com.wfsample.delivery;
+
+import com.wfsample.common.dto.DeliveryStatusDTO;
+import com.wfsample.common.dto.PackedShirtsDTO;
+import com.wfsample.service.DeliveryApi;
+
+import java.util.UUID;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * Controller for delivery service which is responsible for dispatching shirts returning tracking
+ * number for a given order.
+ *
+ * @author Hao Song (songhao@vmware.com).
+ */
+public class DeliveryController implements DeliveryApi {
+
+  @Override
+  public Response dispatch(String orderNum, PackedShirtsDTO packedShirts) {
+    if (orderNum.isEmpty()) {
```

Comment:
Add a randomizer here so that 1 in 5 requests fail to deliver and return Response.Status.BAD_REQUEST